homeassistant/components/spokestack/tts.py

```diff
@@ -0,0 +1,82 @@
+"""Support for the Spokestack Text to Speech Service."""
+import logging
+
+from spokestack.tts.clients.spokestack import TextToSpeechClient, TTSError
+
+from homeassistant.components.tts import CONF_LANG, Provider
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    CONF_IDENTITY,
+    CONF_SECRET_KEY,
+    DEFAULT_IDENTITY,
+    DEFAULT_LANG,
+    DEFAULT_MODE,
+    DEFAULT_PROFILE,
+    DEFAULT_SECRET_KEY,
+    DEFAULT_VOICE,
+    SUPPORTED_LANGUAGES,
+    SUPPORTED_MODES,
+    SUPPORTED_OPTIONS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_engine(hass: HomeAssistant, config, discovery_info=None):
+    """Set up Spokestack TTS provider."""
+    return SpokestackProvider(hass, config)
```

Comment:
You shuold only setup if set up via discovery.
```suggestion
    if discovery_info is None:
        return
    return SpokestackProvider(hass, config)
```

Reply:
I'm a little confused about how `discovery_info` gets passed to this function. It looks like from here `discovery_info` will be `None`

https://github.com/home-assistant/core/blob/80f1e8770aac986e2f231e39447f1da7312ed00f/homeassistant/components/tts/__init__.py#L221